blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeGraphVisitor.java

```diff
@@ -187,16 +189,15 @@ protected void handleChunkDone(@Nonnull MemoryFlowChunk chunk) {
             return;
         }
 
-        // its stage inside parallel, we skip it and clear nest stages collected inside parallel
+        boolean parallelNestedStages = false;
+
+        // it's nested stages inside parallel so let's collect them later
         if(parallelEnd != null){
-            return;
+            parallelNestedStages = true;
         }
 
         if(!nestedStages.empty()){
-            nestedStages.pop(); //we throw away nested stages
```

Comment:
We should verify scripted pipeline behavior. Nested stages in scripted pipeline have been skipped. e.g. below pipeline scripted should not generated nodes in DAG for BranchStage1 and NestedBranchStage1 stages.

```
node {
   stage ('test') { 
     echo ('Testing'); 
     parallel NestedStagesBranch: {
        echo 'first Branch'
        stage('BranchStage1') {
          echo 'NestedBranchStage1'
           stage('NestedBranchStage') {
              echo 'NestedBranchStage1'
            }
        }
        echo 'first Branch end'
     }
   }    
}
```

Reply:
fixed